bgpd/bgpd.c

```diff
@@ -108,7 +108,8 @@ static int bgp_check_main_socket(bool create, struct bgp *bgp)
 	struct listnode *bgpnode, *nbgpnode;
 	struct bgp *bgp_temp;
 
-	if (bgp->inst_type == BGP_INSTANCE_TYPE_VRF)
+	if (bgp->inst_type == BGP_INSTANCE_TYPE_VRF &&
```

Comment:
@pguibert6WIND @vjardin6WIND This is exactly the type of wrong abstraction I've been talking about from a protocol perspective..  I'd really like to sit down and figure this out more.  Can you file a bug to track the vrf protocol abstraction and I'll accept this code in the mean time, after I've had a chance to run another round of tests over night?

Reply:
https://github.com/FRRouting/frr/issues/1845